### PR TITLE
[OSDEV-902] Implement required fields serializer

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-902](https://opensupplyhub.atlassian.net/browse/OSDEV-902) Implement required fields serializer
- Implemented header validation structure + required fields validation

### CODE REVIEW

**Review by killev — CHANGES_REQUESTED** [review]
As basic architecture, I don't suppose to have header parser at all. Instead, I supposed to create RowRequiredFieldsSerializer with implementation. So that it validates incoming row's keys agains list of retired keys in it

File: src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py [hunk 1]
```diff
@@ -17,6 +20,7 @@ def __init__(self):
             RowSectorSerializer(),
             RowCountrySerializer(),
             RowEmptySerializer(),
+            RowRequiredFieldsSerializer(),
```
Comment (killev) [inline src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py:23]: It should go after RowEmptySerializer which is simply add all feeds to the result

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
Left one comment

File: src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class RowRequiredFieldsSerializer(RowSerializer):
+
+    def validate(self, row: dict, current: dict) -> dict:
+        required_fields = ["name",
+                           "address",
+                           "country"]
+
+        for required_field in required_fields:
+            if required_field in row.keys():
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py]: Resolved

**Review by vladsha-dev — APPROVED** [review]
LGTM

File: src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from .row_serializer import RowSerializer
+
+
+class RowRequiredFieldsSerializer(RowSerializer):
```
Comment (killev) [inline src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py:4]: I think. there's a better implementation could be copied from prev closed PR:

```
class HeaderRequredValidator:
     required_fields = {"name", "address", ""}

     def validate(self, raw_header: List[str]) -> dict:
         diff = self.required_fields.difference(raw_header)

         if len(diff) > 0:
             return {
                 "error": {
                     "message": "{} are missing".format(', '.join(diff)),
                     "type": "Error",
                 },
             }
         return {}
```

**Review by killev — CHANGES_REQUESTED** [review]
Better implementation can be applied, please see comment inline

File: src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py (hunk 3 above)
Comment (roman-stolar) [reply src/django/contricleaner/lib/serializers/row_serializers/row_required_fields_serializer.py:4]: Will do

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**